File: romanNumeralsJava/src/main/java/com/codurance/RomanNumerals.java
package com.codurance;

/**
 * Created by tobyretallick on 09/09/2014.
 */
public class RomanNumerals {

    public static String convert(int value) {
        String convertedValue = "";
        for(ConversionTable row : ConversionTable.values()) {
            while(value >= row.arabic) {
                convertedValue += row.roman;
                value -= row.arabic;
            }
        }
        return convertedValue;
    }

    private static enum ConversionTable {
        FIFTY(50, "L"),
        TEN(10, "X"),
        NINE(9, "IX"),
        FIVE(5, "V"),
        FOUR(4, "IV"),
        ONE(1, "I");

        private int arabic;
        private String roman;

        ConversionTable(int arabic, String roman){
            this.arabic = arabic;
            this.roman = roman;
        }
    }
}

File: RomanNumeralsKataRuby/lib/romannumerals.rb
#romannumerals
class RomanNumeral

  SYMBOLS = [
    ["C",100], ["L",50],
    ["XL",40], ["X",10], ["IX",9], ["V",5], ["IV",4], ["I",1]
  ]

  def convert(number)
    result = ""
    
    SYMBOLS.each do | roman, arabic|
      while number >= arabic
        result += roman
        number -= arabic
      end      
    end

    result
  end

end